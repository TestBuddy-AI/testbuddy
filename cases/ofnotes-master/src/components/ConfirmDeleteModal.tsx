import React from "react";
import {
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText,
  DialogActions,
  Button,
  makeStyles,
  createStyles,
  Slide,
  SlideProps,
} from "@material-ui/core";

const useStyles = makeStyles((theme) =>
  createStyles({
    delete: {
      backgroundColor: theme.palette.error.main,
      color: theme.palette.error.contrastText,
    },
  })
);

const Transition = React.forwardRef((props: SlideProps, ref) => (
  <Slide direction="up" ref={ref} {...props} />
));

interface Props {
  open: boolean;
  title: string;
  body?: string;
  onClose(): void;
  onConfirm(): void;
}

export const ConfirmDeleteModal: React.FC<Props> = ({
  open,
  onClose,
  onConfirm,
  title,
  body = null,
}) => {
  const classes = useStyles();

  return (
    <Dialog
      disablePortal
      open={open}
      onClose={onClose}
      maxWidth="sm"
      fullWidth={true}
      TransitionComponent={Transition}
    >
      {title && <DialogTitle>{title}</DialogTitle>}
      {body && (
        <DialogContent>
          <DialogContentText>{body}</DialogContentText>
        </DialogContent>
      )}
      <DialogActions>
        <Button onClick={onClose} color="default">
          Cancel
        </Button>
        <Button
          onClick={onConfirm}
          className={classes.delete}
          variant="contained"
          autoFocus
        >
          Delete
        </Button>
      </DialogActions>
    </Dialog>
  );
};
