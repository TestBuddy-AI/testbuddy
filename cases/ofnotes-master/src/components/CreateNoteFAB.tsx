import React from "react";
import { Link } from "react-router-dom";
import {
  makeStyles,
  createStyles,
  Fab,
  useTheme,
  useMediaQuery,
} from "@material-ui/core";
import Icon from "@material-ui/icons/NoteAdd";

const useStyles = makeStyles((theme) =>
  createStyles({
    root: {
      position: "fixed",
      right: theme.spacing(4),
      bottom: theme.spacing(4),
      "& svg": {
        marginRight: theme.spacing(),
      },

      [theme.breakpoints.down("sm")]: {
        right: theme.spacing(2),
        bottom: theme.spacing(2),
        "& svg": {
          marginRight: 0,
        },
      },
    },
  })
);

export const CreateNoteFAB: React.FC = () => {
  const classes = useStyles();
  const theme = useTheme();
  const collapse = useMediaQuery(theme.breakpoints.down("sm"));

  return (
    <Fab
      classes={{ root: classes.root }}
      variant={collapse ? "round" : "extended"}
      color="primary"
      component={Link}
      to="/create"
      size="large"
    >
      <Icon />
      {!collapse && "New Note"}
    </Fab>
  );
};
