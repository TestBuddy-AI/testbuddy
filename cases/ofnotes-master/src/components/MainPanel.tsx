import React from "react";
import clsx from "clsx";
import { makeStyles, createStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      flexShrink: 1,
      flexBasis: "100%",
      overflowY: "auto",
      position: "relative",
      zIndex: 0,
      padding: theme.spacing(4),
    },
  })
);

interface Props extends Partial<JSX.IntrinsicElements["div"]> {
  className?: string;
}

export const MainPanel = React.forwardRef<HTMLDivElement, Props>(
  ({ className, children, ...rest }, ref) => {
    const classes = useStyles();
    return (
      <div className={clsx(classes.root, className)} ref={ref} {...rest}>
        {children}
      </div>
    );
  }
);
