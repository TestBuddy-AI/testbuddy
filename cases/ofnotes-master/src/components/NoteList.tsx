import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import {
  makeStyles,
  createStyles,
  List,
  ListItem,
  ListItemAvatar,
  Avatar,
  ListItemText,
  ListItemSecondaryAction,
  IconButton,
} from "@material-ui/core";
import EditIcon from "@material-ui/icons/Edit";
import { Note } from "db";

const useStyles = makeStyles(
  createStyles({
    bullet: {
      display: "inline-block",
      margin: "0 3px",
      transform: "scale(0.8)",
    },
  })
);

interface Props {
  notes: Note[];
}

export const NoteList: React.FC<Props> = ({ notes }) => {
  const classes = useStyles();

  const bull = <span className={classes.bullet}>â€¢</span>;

  return (
    <List>
      {notes.map((note) => (
        <ListItem key={note.slug} button component={Link} to={`/${note.slug}`}>
          <ListItemAvatar>
            <Avatar>{note.title.charAt(0)}</Avatar>
          </ListItemAvatar>
          <ListItemText
            primary={note.title}
            primaryTypographyProps={{
              noWrap: false,
            }}
            secondary={
              <span>
                {moment(note.modified).fromNow()}
                {note.tags.length ? bull : ""}
                {note.tags.join(", ")}
              </span>
            }
          />
          <ListItemSecondaryAction>
            <IconButton component={Link} to={`/${note.slug}/edit`}>
              <EditIcon />
            </IconButton>
          </ListItemSecondaryAction>
        </ListItem>
      ))}
    </List>
  );
};
