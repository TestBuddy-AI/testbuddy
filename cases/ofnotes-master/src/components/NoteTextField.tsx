import React from "react";
import clsx from "clsx";
import { makeStyles, createStyles } from "@material-ui/core/styles";
import { TextField, OutlinedTextFieldProps } from "@material-ui/core";

const useStyles = makeStyles((theme) =>
  createStyles({
    inp: {
      marginBottom: theme.spacing(3),
    },
  })
);

export const NoteTextField = ({
  className,
  value,
  name,
  ...rest
}: Partial<OutlinedTextFieldProps>) => {
  const classes = useStyles();
  return (
    <TextField
      id={name}
      name={name}
      variant="outlined"
      fullWidth
      autoComplete="off"
      className={clsx(classes.inp, className)}
      InputLabelProps={{
        shrink: Boolean(value),
      }}
      value={value}
      {...rest}
    />
  );
};
