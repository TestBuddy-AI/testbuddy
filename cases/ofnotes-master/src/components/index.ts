export * from "./SideMenu";
export * from "./NoteForm";
export * from "./Layout";
export * from "./NoteTextField";
export * from "./MainPanel";
export * from "./CreateNoteFAB";
export * from "./SideMenuHeader";
export * from "./SideMenuFilter";
export * from "./NoteList";
export * from "./RenderedNote";
export * from "./NoteForm";
export * from "./ConfirmDeleteModal";
