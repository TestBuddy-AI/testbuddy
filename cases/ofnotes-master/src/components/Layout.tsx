import React from "react";
import { CssBaseline, makeStyles, createStyles } from "@material-ui/core";
import { SideMenu } from "components";

const useStyles = makeStyles((theme) =>
  createStyles({
    root: {
      display: "flex",
      width: "100%",
      height: "100vh",
      overflowX: "hidden",
    },
  })
);

export const Layout: React.FC = ({ children }) => {
  const classes = useStyles();

  return (
    <>
      <CssBaseline />
      <div className={classes.root}>
        <SideMenu />
        {children}
      </div>
    </>
  );
};
