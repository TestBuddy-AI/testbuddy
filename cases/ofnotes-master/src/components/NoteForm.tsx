import React from "react";
import { makeStyles, createStyles } from "@material-ui/core/styles";
import {
  OutlinedTextFieldProps,
  Button,
  InputAdornment,
} from "@material-ui/core";
import green from "@material-ui/core/colors/green";
import yellow from "@material-ui/core/colors/yellow";
import DeleteIcon from "@material-ui/icons/DeleteForever";
import SavedIcon from "@material-ui/icons/Check";
import UnsavedIcon from "@material-ui/icons/HourglassFullRounded";

import { UserNoteFields } from "db";
import { NoteTextField } from "./NoteTextField";

const useStyles = makeStyles((theme) =>
  createStyles({
    root: {
      display: "flex",
      height: "100%",
      flexDirection: "column",
    },
    textarea: {
      height: "calc(100vh - 400px)",
      [theme.breakpoints.up("lg")]: {
        height: "calc(100vh - 360px)",
      },
    },
    progress: {
      position: "absolute",
      top: 0,
    },
    delete: {
      borderColor: theme.palette.error.main,
      color: theme.palette.error.main,
      width: 200,
    },
    "@keyframes spin": {
      from: {
        transform: "rotate(0deg)",
      },
      to: {
        transform: "rotate(360deg)",
      },
    },
    savedIcon: {
      color: green[500],
    },
    unsavedIcon: {
      color: yellow[200],
      // animation: "$spin 2000ms infinite"
    },
  })
);

interface Props extends Omit<Partial<OutlinedTextFieldProps>, "value"> {
  values: UserNoteFields;
  saved?: boolean;
  showDelete?: boolean;
  errors: {
    [K in keyof UserNoteFields]?: string | null;
  };
  onChange: React.ChangeEventHandler<HTMLInputElement>;
  onBlur: React.FocusEventHandler<HTMLInputElement>;
  onDelete?: React.MouseEventHandler<HTMLButtonElement>;
}

export const NoteForm: React.FC<Props> = ({
  values,
  errors,
  showDelete = false,
  onDelete,
  saved,
  ...rest
}) => {
  const classes = useStyles();

  const LoadingAdornment: React.FC = () => (
    <InputAdornment position="end">
      <UnsavedIcon classes={{ root: classes.unsavedIcon }} />
    </InputAdornment>
  );

  const SavedAdornment: React.FC = () => (
    <InputAdornment position="end">
      <SavedIcon classes={{ root: classes.savedIcon }} />
    </InputAdornment>
  );

  const adorn =
    saved === undefined ? undefined : saved ? (
      <SavedAdornment />
    ) : (
      <LoadingAdornment />
    );

  return (
    <div className={classes.root}>
      <NoteTextField
        name="title"
        label="Title"
        error={!!errors.title}
        helperText={errors.title}
        value={values.title}
        InputProps={{
          endAdornment: adorn,
        }}
        {...rest}
      />

      <NoteTextField
        name="md"
        label="Note"
        multiline
        rows={20}
        InputProps={{
          classes: {
            input: classes.textarea,
          },
        }}
        error={!!errors.md}
        helperText={errors.md ?? "Supports github flavored markdown"}
        value={values.md}
        {...rest}
      />

      <NoteTextField
        name="tags"
        label="Tags"
        error={!!errors.tags}
        helperText={errors.tags ?? "Separated by spaces"}
        value={values.tags.join(" ")}
        {...rest}
      />

      {showDelete && (
        <Button
          onClick={onDelete}
          variant="outlined"
          size="small"
          fullWidth={false}
          classes={{ root: classes.delete }}
          startIcon={<DeleteIcon />}
        >
          Delete
        </Button>
      )}
    </div>
  );
};
