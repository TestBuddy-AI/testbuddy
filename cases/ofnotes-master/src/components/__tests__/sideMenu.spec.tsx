import React from "react";
import { render, fireEvent, wait, fakeNotes } from "test-utils";
import { SideMenu } from "components";

describe("SideMenu component", () => {
  const expandedMock = jest.fn();
  const filterChangeMock = jest.fn();

  const defaultSidebarCtx = {
    expanded: true,
    setExpanded: expandedMock,
    filter: "",
    onFilterChange: filterChangeMock,
  };

  afterEach(() => {
    expandedMock.mockReset();
    filterChangeMock.mockReset();
  });

  it("matches snapshot", async () => {
    const { asFragment, getByText } = render(<SideMenu />, {
      sideBarCtx: {
        ...defaultSidebarCtx,
      },
      noteCtx: {
        notes: fakeNotes,
      },
    });
    await wait(() => {
      expect(getByText(fakeNotes[0].title)).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("sets expanded to true on swipe right", async () => {
    const { container } = render(<SideMenu />, {
      sideBarCtx: {
        ...defaultSidebarCtx,
        expanded: true,
      },
    });
    const element = container.querySelector("div.MuiPaper-root") as Element;
    fireEvent.touchStart(element, {
      touches: [{ clientX: 5, clientY: 10 }],
    });
    fireEvent.touchMove(element, {
      touches: [{ clientX: 25, clientY: 10 }],
    });
    fireEvent.touchEnd(element);
    await wait(() => {
      expect(expandedMock).toHaveBeenCalled();
      expect(expandedMock.mock.calls[0][0]).toEqual(true);
    });
  });

  it("sets expanded to false on swipe left", async () => {
    const { container } = render(<SideMenu />, {
      sideBarCtx: {
        ...defaultSidebarCtx,
        expanded: true,
      },
    });
    const element = container.querySelector("div.MuiPaper-root") as Element;
    fireEvent.touchStart(element, {
      touches: [{ clientX: 25, clientY: 10 }],
    });
    fireEvent.touchMove(element, {
      touches: [{ clientX: 5, clientY: 10 }],
    });
    fireEvent.touchEnd(element);
    await wait(() => {
      expect(expandedMock).toHaveBeenCalled();
      expect(expandedMock.mock.calls[0][0]).toEqual(false);
    });
  });

  it("can swipe hidden swipe handler", async () => {
    const { container } = render(<SideMenu />, {
      sideBarCtx: {
        ...defaultSidebarCtx,
        expanded: false,
      },
    });
    const element = container.querySelectorAll("div")[0] as Element;
    fireEvent.touchStart(element, {
      touches: [{ clientX: 5, clientY: 10 }],
    });
    fireEvent.touchMove(element, {
      touches: [{ clientX: 25, clientY: 10 }],
    });
    fireEvent.touchEnd(element);
    await wait(() => {
      expect(expandedMock).toHaveBeenCalled();
      expect(expandedMock.mock.calls[0][0]).toEqual(true);
    });
  });

  it("sets expanded to false on close menu click", async () => {
    const { getByText } = render(<SideMenu />, {
      sideBarCtx: {
        ...defaultSidebarCtx,
        expanded: true,
      },
    });
    fireEvent.click(getByText("close menu"));
    await wait(() => {
      expect(expandedMock).toHaveBeenCalled();
      expect(expandedMock.mock.calls[0][0]).toEqual(false);
    });
  });

  it("sets expanded to true on open menu click", async () => {
    const { getByText } = render(<SideMenu />, {
      sideBarCtx: {
        ...defaultSidebarCtx,
        expanded: false,
      },
    });
    fireEvent.click(getByText("open menu"));
    await wait(() => {
      expect(expandedMock).toHaveBeenCalled();
      expect(expandedMock.mock.calls[0][0]).toEqual(true);
    });
  });
});
