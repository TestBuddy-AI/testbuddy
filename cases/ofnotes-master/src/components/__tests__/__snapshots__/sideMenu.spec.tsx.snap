// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root makeStyles-container-1 makeStyles-container-3 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="makeStyles-root-4 makeStyles-root-6"
    >
      <div
        class="makeStyles-header-5"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          OfNote
          <small>
            <em>
              s
            </em>
          </small>
        </h5>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
        >
          Offline note taking
        </h6>
      </div>
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root"
        href="https://github.com/jhackshaw/ofnotes"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            class="MuiSvgIcon-root"
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 15.31L23.31 12 20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6z"
            />
            <title>
              set dark theme
            </title>
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            class="MuiSvgIcon-root"
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <title>
              close menu
            </title>
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="makeStyles-root-7"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inp-8 MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
          data-shrink="true"
          for="filter"
          id="filter-label"
        >
          Filter
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-describedby="filter-helper-text"
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="filter"
            name="filter"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
            >
              <span>
                Filter
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
          id="filter-helper-text"
        >
          Search tags or titles
        </p>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <ul
      class="MuiList-root MuiList-padding"
    >
      <li
        class="MuiListItem-container"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
          href="/test-one"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemAvatar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              t
            </div>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              test one
            </span>
            <p
              class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              <span>
                a few seconds ago
              </span>
            </p>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <div
          class="MuiListItemSecondaryAction-root"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root"
            href="/test-one/edit"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </li>
      <li
        class="MuiListItem-container"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
          href="/test-two"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemAvatar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              t
            </div>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              test two
            </span>
            <p
              class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              <span>
                2 hours ago
                <span
                  class="makeStyles-bullet-13"
                >
                  â€¢
                </span>
                python
              </span>
            </p>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <div
          class="MuiListItemSecondaryAction-root"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root"
            href="/test-two/edit"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </li>
      <li
        class="MuiListItem-container"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
          href="/test-three"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemAvatar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              t
            </div>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              test three
            </span>
            <p
              class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              <span>
                22 days ago
                <span
                  class="makeStyles-bullet-13"
                >
                  â€¢
                </span>
                test, three
              </span>
            </p>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <div
          class="MuiListItemSecondaryAction-root"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root"
            href="/test-three/edit"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
